File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.js
/**
 * @param {number[][]} grid
 * @param {number} x
 * @return {number}
 */
var minOperations = function(grid, x) {
    
    if(grid.length == 1 && grid[0].length == 0) return 0;
    const arr = new Array();
    let count = 0;

    for(const row of grid) {

        for(const col of row) {

             arr.push(col);
            
        }
    }

    arr.sort((a,b) => a - b);
    let target = arr[Math.floor(arr.length / 2)];

    for(const num of arr) {

        let t = Math.abs(target - num);
        if((t % x) != 0 ) return -1;
        count += ( t / x);
        
    }

    return count;
};

File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.cpp
class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x) {
        
        vector<int> arr;

        for(auto row: grid) {
            for(auto col: row) {
                arr.push_back(col);
            }
        }

        sort(arr.begin(), arr.end());

    
        int mid = arr.size()  / 2;
        int count = 0;

        int midValue = arr[mid];

        for(int i = 0; i < arr.size(); i++ ) {
            int rem = abs(arr[i] - midValue);
            
            if(rem % x != 0) return -1;
            count += rem / x;
        }
        return count;
    }

};